### PR TITLE
Migrate Angola holidays to holiday groups

### PR DESCRIPTION
**Pull request by kasya**
## Proposed change

Migrate Angola holidays to holiday groups

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/countries/angola.py [hunk 1]
```diff
@@ -9,79 +9,119 @@
...
         if year >= 1980:
-            _add_with_observed(date(year, SEP, 17), "Dia do Herói Nacional")
+            self._add_observed_holiday(
+                self._add_holiday("Dia do Herói Nacional", SEP, 17)
+            )
+
+        # All Souls' Day.
+        self._add_observed_holiday(
+            self._add_holiday("Dia dos Finados", NOV, 2)
+        )
```
Comment (KJhellico) [inline holidays/countries/angola.py:114]:
```suggestion
        self._add_observed_holiday(
            self._add_all_souls_day("Dia dos Finados")
        )
```

**Review by KJhellico — COMMENTED** [review]
LGTM 👍

**Review by arkid15r — COMMENTED** [review]
Marking as ready for beta 👍

**Comment by KJhellico** [comment]
It has been merged into beta. Thank you for the contribution! 👍

**Pull request merged**